substrate/primitives/core/Cargo.toml

```diff
@@ -132,6 +132,7 @@ serde = [
 	"secrecy/alloc",
 	"sp-core-hashing",
 	"sp-storage/serde",
+	"sp-runtime-interface/disable_target_static_assertions",
```

Comment:
This probably should not be here. W/o compilation fails:
```
error[E0512]: cannot transmute between types of different sizes, or dependently-sized types
  --> substrate/primitives/runtime-interface/src/impls.rs:45:1
   |
45 | assert_eq_size!(usize, u32);
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: source type: `usize` (64 bits)
   = note: target type: `u32` (32 bits)
   = note: this error originates in the macro `assert_eq_size` (in Nightly builds, run with -Z macro-backtrace for more info)
```

Reply:
Substrate currently assumed that `no_std == arch::wasm32`. Shitty, I know, but that is currently the assumption.

Reply:
Yes, that assert should be changed to check for `wasm32` and not for `no_std`, and then you won't need to disable it.

...as Basti said, the current assumption everywhere is that no_std equals WASM, which, uh, I guess I'll have to fix soon-ish since I'll be wanting to get the RISC-V based runtimes working. :P

Reply:
Yeah good idea.